syntax = "proto3";

package unsloth;

service UnslothModelService {
    rpc Generate(GenerateRequest) returns (stream GenerateResponse);
    rpc Init(InitRequest) returns (InitResponse);
    rpc UnloadLLM(UnloadLLMRequest) returns (UnloadLLMResponse);
}

// The request message for the chat.
message GenerateRequest {
    string prompt = 1;
}

message GenerateResponse {
    string generated_text = 1;
    int32 input_tokens_count = 2;
    int32 output_tokens_count = 3;
}

// The request message for Init.
message InitRequest {
    string model = 1;
}

message InitResponse {
    bool init_status = 1;
}

message UnloadLLMRequest{}

message UnloadLLMResponse{
    bool status = 1;
}